dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance20/impl/FutureShellTest.java

```diff
@@ -0,0 +1,262 @@
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testCancellationCallbackFires() {
+        String value = "testCancellationCallbackFires";
+        final Set<String> callbackResults = new HashSet<String>();
+        Consumer<Boolean> cancellationCallback = createCancellationCallback(value, callbackResults);
+        FutureShell<String> futureShell = new FutureShell<String>();
+
+        futureShell.setCancellationCallback(cancellationCallback);
+        futureShell.cancel(false);
+
+        assertTrue(callbackResults.contains(value));
+    }
+
+    @Test //This test is a known failure.
+    public void testCancellationCallbackWontFireWithDeligate() {
+        String value = "testCancellationCallbackWontFireWithDeligate";
+        final Set<String> callbackResults = new HashSet<String>();
+        Consumer<Boolean> cancellationCallback = createCancellationCallback(value, callbackResults);
+        FutureShell<String> futureShell = new FutureShell<String>();
+
+        try {
+            futureShell.setDelegate(createFuture(value));
+            Thread.sleep(10);
+            futureShell.setCancellationCallback(cancellationCallback);
+            futureShell.cancel(false);
+
+            assertFalse(callbackResults.contains(value));
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testIsDone() {
+        try {
+            FutureShell<String> cancelFutureShell = new FutureShell<String>();
+            assertFalse("Is done returned true with no deligate", cancelFutureShell.isDone());
+
+            cancelFutureShell.cancel(false);
+            assertTrue("isDone returned false after cancelation", cancelFutureShell.isDone());
+
+            FutureShell<String> futureShell = new FutureShell<String>();
+
+            futureShell.setDelegate(createFuture("testIsDone"));
+            assertFalse("Is done returned true when the deligate should still be running", futureShell.isDone());
+            Thread.sleep(510);
+            assertTrue("Is done returned false when the deligate should be done", futureShell.isDone());
+
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testGetWithTimeout() {
+        boolean timedOut = false;
+        try {
+            FutureShell<String> futureShell = new FutureShell<String>();
+            futureShell.setDelegate(createFuture("testGetWithTimeout"));
+            futureShell.get(100, TimeUnit.MILLISECONDS);
+            fail("FutureShell did not throw a timeout exception");
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        } catch (TimeoutException e) {
+            timedOut = true;
+        }
+        assertTrue("No TimeoutException was thrown", timedOut);
```

Comment:
If you allow other exceptions to be thrown, you shouldn't actually need this assert, further simplifying the test.